src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,90 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLICATION_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    MyHashMap() {
+        table = new Node[INITIAL_DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int bucket = getHash(key);
+        Node<K, V> currentNode = table[bucket];
+        Node<K, V> newNode = new Node<>(key, value, null);
+
+        while (currentNode != null) {
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
```

Comment:
Use of '==' for object comparison is incorrect when checking if keys are equal. It should be `Objects.equals(key, currentNode.key)` to handle both `null` and non-null cases correctly.